exp/controllers/rosanetwork_controller.go

```diff
@@ -0,0 +1,328 @@
+				expinfrav1.ROSANetworkReadyCondition,
+				expinfrav1.ROSANetworkFailedReason,
+				clusterv1.ConditionSeverityError,
+				"%s",
+				err.Error())
+			return ctrl.Result{}, fmt.Errorf("failed to start CF stack creation: %w", err)
+		}
+		conditions.MarkFalse(rosaNetScope.ROSANetwork,
+			expinfrav1.ROSANetworkReadyCondition,
+			expinfrav1.ROSANetworkCreatingReason,
+			clusterv1.ConditionSeverityInfo,
+			"")
+		return ctrl.Result{}, nil
+	}
+	// The cloudformation stack already exists
+	if err := r.updateROSANetworkResources(ctx, rosaNetScope.ROSANetwork); err != nil {
+		return ctrl.Result{RequeueAfter: time.Second * 60}, fmt.Errorf("error fetching CF stack resources: %w", err)
+	}
+
+	switch cfStack.StackStatus {
+	case cloudformationtypes.StackStatusCreateInProgress: // Create in progress
+		// Set the reason of false ROSANetworkReadyCondition to Creating
+		conditions.MarkFalse(rosaNetScope.ROSANetwork,
+			expinfrav1.ROSANetworkReadyCondition,
+			expinfrav1.ROSANetworkCreatingReason,
+			clusterv1.ConditionSeverityInfo,
+			"")
+		return ctrl.Result{RequeueAfter: time.Second * 60}, nil
+	case cloudformationtypes.StackStatusCreateComplete: // Create complete
+		if err := r.parseSubnets(rosaNetScope.ROSANetwork); err != nil {
+			return ctrl.Result{}, fmt.Errorf("parsing stack subnets failed: %w", err)
+		}
+
+		// Set the reason of true ROSANetworkReadyCondition to Created
+		// We have to use conditions.Set(), since conditions.MarkTrue() does not support setting reason
+		conditions.Set(rosaNetScope.ROSANetwork,
+			&clusterv1.Condition{
+				Type:     expinfrav1.ROSANetworkReadyCondition,
+				Status:   corev1.ConditionTrue,
+				Reason:   expinfrav1.ROSANetworkCreatedReason,
+				Severity: clusterv1.ConditionSeverityInfo,
+			})
+		return ctrl.Result{}, nil
+	case cloudformationtypes.StackStatusCreateFailed: // Create failed
+		// Set the reason of false ROSANetworkReadyCondition to Failed
+		conditions.MarkFalse(rosaNetScope.ROSANetwork,
+			expinfrav1.ROSANetworkReadyCondition,
+			expinfrav1.ROSANetworkFailedReason,
+			clusterv1.ConditionSeverityError,
+			"")
+		return ctrl.Result{}, fmt.Errorf("cloudformation stack %s creation failed, see the stack resources for more information", *cfStack.StackName)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ROSANetworkReconciler) reconcileDelete(ctx context.Context, rosaNetScope *scope.ROSANetworkScope) (res ctrl.Result, reterr error) {
+	rosaNetScope.Info("Reconciling ROSANetwork delete")
+
+	// Try to fetch CF stack with a given name
+	stackName := rosaNetScope.ROSANetwork.Spec.Name
+	cfStack, err := r.awsClient.GetCFStack(ctx, stackName)
+	if err != nil {
+		var apiErr smithy.APIError // in case the stack does not exist, AWS returns ValidationError
+		if errors.As(err, &apiErr) && apiErr.ErrorCode() == "ValidationError" {
+			cfStack = nil
+		} else {
+			return ctrl.Result{}, fmt.Errorf("error fetching CF stack details: %w", err)
+		}
+	}
```

Comment:
This part is duplicated reconcileDelete & reconcileNormal funcs better to move it to Reconcile func

Reply:
Yep, moved.